### DISCUSSION TITLE
Consider another storage model for translations

### DISCUSSION THREAD

**Discussion by ax1mx2** [payloadcms/payload#1066] [General]
Hi,

We have a large database with products in many languages (~ 50). How do you handle the case were one reaches the document size limit of 16MiB in MongoDB? So far, contents in different languages is stored in a map where the key is the locale identifier.

Are there any plans to support content items with unlimited number of languages? (Possibly with another database or database model in MongoDB.)

Resources:
https://www.mongodb.com/docs/manual/reference/limits/#mongodb-limit-BSON-Document-Size

**Comment by jmikrut** [comment]
Oh man - you're hitting the 16mb limit? You may be the first! Yeah, we do have additional database support on our roadmap, but I think for your use case, you may want to consider re-architecting your schema with the tools that Payload already provides. Is there any opportunity for that? Because even 16mb / 50 locales - those are some big documents!
